### DISCUSSION TITLE
How to implement security/sanitization for ~/server/api?

### DISCUSSION THREAD

**Discussion by john-prutton** [nuxt/nuxt#16983] [Questions] [answered]
Hi there, a Nuxt newcomer here!

 I am switching from `Express` to Nuxt's `~/server/` and am wondering about `H3`'s security and middleware. Some  middleware that was on the Express app is [Helmet](https://helmetjs.github.io/), [Mongo-sanitize](https://www.npmjs.com/package/mongo-sanitize), [XSS](https://www.npmjs.com/package/xss), and [HPP](https://www.npmjs.com/package/hpp).

Some of these middleware are simple enough that I feel like I could implement them myself, like Helmet as it seems it just sets headers, but I would much prefer if I could use these as server middleware, or, even better, not have to use them - if H3 has these features built in.

**How would I go about implementing these middleware in my Nuxt 3 application? Should I sanitize data inside `~/server/api/` routes by hand? Did I miss something on the [H3 docs](https://www.jsdocs.io/package/h3)? Should I use Express inside `~/server/index.ts`?**

Thanks in advance for any help or recommendations!

**Comment by TheAlexLichter** [comment, marked as answer]
You can use most Express middleware with H3 as `serverMiddleware`. See https://github.com/wobsoriano/h3-session/blob/master/src/index.ts as example. ☺️
